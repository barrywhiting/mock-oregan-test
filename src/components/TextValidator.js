import '../styles/TextValidator.css';

import React from 'react';

const TextValidator = (props) => {
    // Just added this to display a nice message to the user to let them know their 
    // choice is invalid.  For the password field I think it would be nicer to give a better message
    // listing what is actually needed for a valid password but at least there is something in 
    // place to help.
    return(
        <div
            id={`${props.id}error`}
            className={`errorText ${(props.id.includes('userName') &&
                props.emailValid) ||
                (props.id.includes('password') &&
                    props.passwordValid) ? 'hideError' : 'showError'}`}
        >{props.errorMessage}</div>

    )
}

export default TextValidator;