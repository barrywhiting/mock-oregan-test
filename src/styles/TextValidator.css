.errorText {
    display: block;
    color: #ff0000;
    height: 0.6rem;
    margin-left: 0.2rem;
    padding-top: 0.2rem;
    padding-bottom: 0.6rem;
    clear: both;
}

.hideError {
    display: none;
}

.showError {
    display: block;
}

.passwordError {
    padding-top: 1.8rem !important;
    padding-bottom: 1.8rem !important;
}

#password1error {
    height: 0.8rem;
}