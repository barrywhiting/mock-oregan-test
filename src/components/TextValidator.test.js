import React from "react";
import TextValidator from './TextValidator';
import renderer from 'react-test-renderer';

describe("Snapshot test for the TextValidator component", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<TextValidator />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
