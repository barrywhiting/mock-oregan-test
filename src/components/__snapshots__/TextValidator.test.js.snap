// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the TextValidator component Matches DOM Snapshot 1`] = `null`;
